### DISCUSSION TITLE
MPD cannot see my music and playlist folders

### DISCUSSION THREAD

**Discussion by ghost** [MusicPlayerDaemon/MPD#2091] [General]
might be best explained with some terminal output (see below)

Any ideas on how to fix? I'm on Fedora 40.

-----------------------

$ sudo chmod -R 777 /mnt/storage/my-stuff/music/lossless

$ sudo chmod -R 777 /mnt/storage/my-stuff/music/playlists

$ sudo systemctl stop mpd

$ sudo systemctl start mpd

$ sudo systemctl status mpd
● mpd.service - Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: active (running) since Thu 2024-07-31 11:22:45 ACST; 5s ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
   Main PID: 30302 (mpd)
      Tasks: 3 (limit: 77060)
     Memory: 11.5M (peak: 12.4M)
        CPU: 266ms
     CGroup: /system.slice/mpd.service
             └─30302 /usr/bin/mpd --systemd

Jul 31 11:22:45 my-desktop1 systemd[1]: Starting mpd.service - Music Player Daemon...
Jul 31 11:22:45 my-desktop1 mpd[30302]: Jul 31 11:44 : exception: Failed to access /mnt/storage/my-stuff/music/playlists: Permission denied
Jul 31 11:22:45 my-desktop1 mpd[30302]: Jul 31 11:44 : exception: Failed to access /mnt/storage/my-stuff/music/lossless: Permission denied
Jul 31 11:22:45 my-desktop1 mpd[30302]: Failed to initialize io_uring: io_uring_queue_init() failed: Permission denied
Jul 31 11:22:45 my-desktop1 systemd[1]: Started mpd.service - Music Player Daemon.

$ sudo grep -i "user" /etc/mpd.conf
user                    "mpd"

**Comment by MaxKellermann** [comment]
This isn't the place to learn about basic Linux permissions because this has nothing to do with MPD - but you should know that you need "execute" permissions on all parent directories. And you should one day learn that mode "777" is an extremely bad idea.

**Reply by ghost** [reply]
1. I only only used "777" just to make sure MPD has read/write access (in case it needs write access).
2. I need to confirm later but I believe the default folder permissions (beginning at /mnt) is 755. and files was set to 644

**Reply by ghost** [reply]
I just checked and I can confirm that this path (all the way from /mnt to music folder) has a folder permission of 755 already.
